ARCHITECTURE.md

```diff
@@ -0,0 +1,226 @@
+[Go structs](https://github.com/istio/operator/blob/master/pkg/apis/istio/v1alpha2/values_types.go).
+
+## The new API
+
+The new API is intended to replace the installation and K8s parts of Helm values.yaml.
+
+### Features and components
+
+The operator has a very similar structure to istio/installer: components are grouped into features.
+The operator API defines functional settings at the feature level. Functional settings are those that performs some
+function in the Istio control plane without necessarily being tied to any one component that runs in a Deployment.
+Component settings are those that necessarily refer to a particular Deployment or Service. For example, the number
+of Pilot replicas is clearly a component setting, because it refers to a component which is a Deployment in the
+cluster. Most K8s platform settings are necessarily component settings.
+The available features and components are as follows:
+
+| Feature | Components |
+|---------|------------|
+Base | CRDs
+Traffic Management | Pilot
+Policy | Policy
+Telemetry | Telemetry
+Security | Citadel
+| | Node agent
+| | Cert manager
+Configuration management | Galley
+Gateways | Ingress gateway
+| | Egress gateway
+AutoInjection | Sidecar injector
+
+Features and components are defined in the
+[name](https://github.com/istio/operator/blob/905dd84e868a0b88c08d95b7ccf14d085d9a6f6b/pkg/name/name.go#L38) package.
+
+### Namespaces
+
+The operator API and underlying Helm charts offer a lot of flexibility in which namespaces features and components are
+installed into. Namespace definitions can be defined and specialized at the global, feature and component level, with
+each lower level overriding the setting of the higher parent level. For example, if the global default namespace
+is defined as:
+
+```yaml
+defaultNamespacePrefix: istio-system
+```
+
+and namespaces are specialized for the security feature and one of the components:
+
+```yaml
+security:
+  components:
+    namespace: istio-security
+    citadel:
+    nodeAgent:
+      namespace: istio-security-nodeagent
+policy:
+  components:
+    policy:
+```
+
+the resulting namespaces will be:
+
+|Component|Namespace|
+|---------|:--------|
+policy | istio-system
+citadel | istio-security
+nodeAgent | istio-security-nodeagent
+
+These rules are expressed in code in the
+[name](https://github.com/istio/operator/blob/905dd84e868a0b88c08d95b7ccf14d085d9a6f6b/pkg/name/name.go#L38) package.
+
+### Enablement
+
+Features and components can be individually or collectively enabled or disabled. If a feature is disabled, all of its
+components are disabled, regardless of their component level enablement. If a feature is enabled, all of its components
+are enabled, unless they individually disabled. For example:
+
+```yaml
+security:
+  enabled: true
+  components:
+    citadel:
+      enabled: false
+```
+
+will enable all components of the Security feature except citadel.
```

Comment:
lowercase "Security".  I am not sure if citadel should be capitalized - might ask @rcaballeromx or @geeknoid.

Reply:
Done.